src/cargo/core/source/source_id.rs

```diff
@@ -247,6 +249,8 @@ impl SourceId {
     }
 
     /// Is this source from an alternative registry
+    /// DEPRECATED: This is not correct if the registry name is not known
+    /// (for example when loaded from an index).
```

Comment:
To confirm, is this left in for clients of the `cargo` crate? Or does Cargo still use this a lot internally?

(if it's the former I think we can just delete this)

Reply:
There is one usage in cargo. I believe it is currently correct due to where it is used, but I put it on my todo list to remove it in the future since it is easy to get wrong.  Its removal isn't trivial, and was unrelated to this PR.